hotel-service/src/main/java/edu/myhotel/hotel/service/storage/DataStorageStub.java

```diff
@@ -195,4 +244,16 @@ private boolean isDateOverlapping(LocalDate start1, LocalDate end1, LocalDate st
     public void bookRoom(int roomId, LocalDate checkIn, LocalDate checkOut) {
         bookedRooms.add(new BookedRoom(roomId, checkIn, checkOut));
     }
+
+    @Override
+    public Room getRoomById(int roomId) {
+        Room current = null;
+        for (Room room : rooms) {
+            if (roomId == room.getId()) {
+                current = room;
+                break;
+            }
+        }
+        return current;
```

Comment:
You can just return the found result from instead of breaking the loop.
```
for (Room room : rooms) {
    if (roomId == room.getId()) {
        return room;
    }
}
return null;
```

It is a kind of search pattern: return a value as soon as it found, otherwise return a default value or throw an exception.

Benefits:
• Little bit shorter
• You clearly see what is a default value
• The algorithm is easier to read and understand – no transition/state (there is no variable that will be changed in some cases)